custom_reports/http_api/definitions/query_request.yaml

```diff
@@ -0,0 +1,123 @@
+type: object
+properties:
+  website_id:
+    type: string
+    format: uuid
+    description: ID of the website.
+    example: 70236925-724f-404e-b634-7b5633bc0bcc
+  columns:
+    type: array
+    description: Column definitions for the query.
+    items:
+      type: object
+      properties:
+        column_id:
+          type: string
+          description: ID of the column (either a dimension or a metric).
+        transformation_id:
+          type: string
+          description: |
+            Optional ID of the transformation. Must be supported by the
+            column that was specified.
+        goal_id:
+          type: integer
+          description: |
+            Optional ID of a goal. Available only for metrics related to goals.
+      required:
+        - column_id
+    example: [{"column_id":"device_type"},{"column_id":"revenue","transformation_id":"sum"}]
+  date_from:
+    type: string
+    format: date
+    description: Start date for the query (inclusive).
+  date_to:
+    type: string
+    format: date
+    description: End date for the query (inclusive).
+  filters:
+    type: object
+    description: |
+      Data filters for the query. Please note that filtering is only available
+      on dimensions at this point. The top-level object must specify AND/OR
+      clause, but any nested objects may either specify another AND/OR
+      clause or a single filter.
+    properties:
+      operator:
+        type: string
+        description: Logical operator (AND/OR) for the clause.
+        enum:
+          - and
+          - or
+      conditions:
+        type: array
+        description: List of filters or recursively nested clauses.
+        items:
+          type: object
+          required:
+            - column_id
+            - condition
+          properties:
+            column_id:
+              type: string
+              description: ID of the column (must be a dimension).
+            transformation_id:
+              type: string
+              description: |
+                Optional ID of the transformation. Must be supported by
+                the column that was specified and must convert the column
+                into another dimension (and not a metric).
+            condition:
+              type: object
+              description: Definition of a the condition for this column.
+              properties:
+                operator:
+                  type: string
+                  description: |
+                    <p>Filtering operator. Available choices depend on
```

Comment:
Formatting of text in OpenAPI is tricky singe we use 2 plugins to render it and both of them render it using different parser (Redoc - Markdown, OpenAPI - Sphinx).

HTML will be passed by makdown, but Sphinx will escape it. Try using formatting that works in both if you can.
(e.g. http://docutils.sourceforge.net/docs/user/rst/quickref.html#paragraphs)

Reply:
removed html